pkg/devfile/parser/parse.go

```diff
@@ -53,60 +54,99 @@ func parseDevfile(d DevfileObj, flattenedDevfile bool) (DevfileObj, error) {
 	return d, nil
 }
 
-// Parse func populates the flattened devfile data, parses and validates the devfile integrity.
+type ParserArgs struct {
+	path             string
+	url              string
+	data             []byte
+	flattenedDevfile *bool // default to true
```

Comment:
a simple boolean value will be default to false, `false` and not provided will be both false
only pointer can be checked if it's `nil`(not provided), or `&false`, so it can be default to true (if == nil).